### PR TITLE
Fixing compilation warnings on SystemZ

### PR DESCRIPTION
**Pull request by DurzaShade**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

SystemZ had compilation warnings. This PR partially addresses the warnings.

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/debug/p/debug_native.c [hunk 1]
```diff
@@ -1450,7 +1450,7 @@ static int rz_debug_native_bp(RzBreakpoint *bp, RzBreakpointItem *b, bool set) {
 		return set
 			? arm32_hwbp_add(dbg, bp, b)
 			: arm32_hwbp_del(dbg, bp, b);
-#elif __riscv
+#elif __riscv || __s390__ || __s390x__ || __zarch__
```
Comment (DurzaShade) [inline librz/debug/p/debug_native.c:1453]: Added the pre-processor macros for SystemZ as per https://sourceforge.net/p/predef/wiki/Architectures/

**Comment by XVilka** [comment]
To be honest, I expected the fix the "right way" - adding the support instead of disabling it.

**Comment by DurzaShade** [comment]
> To be honest, I expected the fix the "right way" - adding the support instead of disabling it.

this is currently just a placeholder PR, that I raised for my GSoC application. I am planning to update it with the fix for https://github.com/rizinorg/rizin/issues/297

**Comment by XVilka** [comment]
I am closing this. If you plan to do the proper fix - please open a new PR.

**Pull request closed without merging**